hydra-configs-torch/tests/test_instantiate_data.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import pytest
+from hydra.utils import get_class, instantiate
+from omegaconf import OmegaConf
+
+import torch.utils.data as data
+
+import torch
+from typing import Any
+
+dummy_tensor = torch.tensor((1, 1))
+dummy_dataset = data.dataset.TensorDataset(dummy_tensor)
+dummy_sampler = data.Sampler(data_source=dummy_dataset)
+
+
+@pytest.mark.parametrize(
+    "modulepath, classname, cfg, passthrough_args, passthrough_kwargs, expected",
+    [
+        pytest.param(
+            "utils.data.dataloader",
+            "DataLoader",
+            {"batch_size": 4},
+            [],
+            {"dataset": dummy_dataset},
+            data.DataLoader(batch_size=4, dataset=dummy_dataset),
+            id="DataLoaderConf",
+        ),
+        pytest.param(
+            "utils.data.dataset",
+            "Dataset",
+            {},
+            [],
+            {},
+            data.Dataset(),
+            id="DatasetConf",
+        ),
+        pytest.param(
+            "utils.data.dataset",
+            "ChainDataset",
+            {},
+            [],
+            {"datasets": [dummy_dataset, dummy_dataset]},
+            data.ChainDataset(datasets=[dummy_dataset, dummy_dataset]),
+            id="ChainDatasetConf",
+        ),
+        pytest.param(
+            "utils.data.dataset",
+            "ConcatDataset",
+            {},
+            [],
+            {"datasets": [dummy_dataset, dummy_dataset]},
+            data.ConcatDataset(datasets=[dummy_dataset, dummy_dataset]),
+            id="ConcatDatasetConf",
+        ),
+        pytest.param(
+            "utils.data.dataset",
+            "IterableDataset",
+            {},
+            [],
+            {},
+            data.IterableDataset(),
+            id="IterableDatasetConf",
+        ),
+        # TODO: investigate asterisk in signature instantiation limitation
```

Comment:
Can you file an issue on the repo so we don't lost track?

Reply:
You saw the rest of the issues, but here:
https://github.com/facebookresearch/hydra/issues/1243

Will get the minimal repro for this one. The others, once we nail them down, we can get minimal repros for those as well.

Reply:
There is a chance that this will one will be difficult to solve. Let's assume we need a workaround.